esp/esdllib/esdl_script.cpp

```diff
@@ -29,6 +29,61 @@
+    virtual IXpathContext* getCopiedSectionXpathContext(IXpathContext* parent, const char* tgtSection, const char* srcSection, bool strictParameterDeclaration) override { return docModel->getCopiedSectionXpathContext(parent, tgtSection, srcSection, strictParameterDeclaration); }
+    virtual void* queryUserData() override { return this; }
+    virtual void setContent(const char* section, const char* xml) override { docModel->setContent(section, xml); }
+    virtual void appendContent(const char* section, const char* name, const char* xml) override { docModel->appendContent(section, name, xml); }
+    virtual void setContent(const char* section, IPropertyTree* tree) override { docModel->setContent(section, tree); }
+    virtual bool tokenize(const char* str, const char* delimiters, StringBuffer& resultPath) override { return docModel->tokenize(str, delimiters, resultPath); }
+    virtual void setAttribute(const char* section, const char* name, const char* value) override { docModel->setAttribute(section, name, value); }
+    virtual const char* queryAttribute(const char* section, const char* name) override { return docModel->queryAttribute(section, name); }
+    virtual const char* getAttribute(const char* section, const char* name, StringBuffer& s) override { return docModel->getAttribute(section, name, s); }
+    virtual const char* getXPathString(const char* xpath, StringBuffer& s) const override { return docModel->getXPathString(xpath, s); }
+    virtual __int64 getXPathInt64(const char* xpath, __int64 dft = false) const override { return docModel->getXPathInt64(xpath, dft); }
+    virtual bool getXPathBool(const char* xpath, bool dft = false) const override { return docModel->getXPathBool(xpath, dft); }
+    virtual void toXML(StringBuffer& xml, const char* section, bool includeParentNode = false) override { docModel->toXML(xml, section, includeParentNode); }
+    virtual void toXML(StringBuffer& xml) override { docModel->toXML(xml); }
+    virtual IPropertyTree* createPTreeFromSection(const char* section) override { return docModel->createPTreeFromSection(section); }
+    virtual void cleanupTemporaries() override { docModel->cleanupTemporaries(); }
+public: // IEsdlScriptContext
+    virtual IEspContext* queryEspContext() const override { return espCtx; }
+    virtual IEsdlFunctionRegister* queryFunctionRegister() const override { return functionRegister; }
+    virtual void setTraceToStdout(bool val) override
+    {
+        if (val != traceToStdout)
+        {
+            traceToStdout = val;
+        }
+    }
+    virtual bool getTraceToStdout() const override { return traceToStdout; }
+    virtual void setTestMode(bool val) override { testMode = val; }
+    virtual bool getTestMode() const override { return testMode; }
+private:
+    Owned<IEspContext>            espCtx;
+    IEsdlFunctionRegister*        functionRegister = nullptr;
+    Owned<ISectionalXmlDocModel>  docModel;
```

Comment:
I'm curious why is this a delegation object rather than using inheritance.  Especially since the corresponding interfaces are using inheritance.

Reply:
Is it because we still pretend we support Xalan as well?

Reply:
That is exactly why. The organization of xmllib does not allow me to know what the base class should be. If the script context was defined in xmllib, where it does not logically belong, it would be inheritance. Now, when we could theoretically have alternate implementations of the model, shouldn't be tied to any one implementation.

Reply:
If I remember correctly, it is the Windows client tools build that still pretends we support Xalan. Something was building with Xalan instead of LibXML2 and the build broke when an interface change was made just to the LibXML2 implementation.

Reply:
Ok.  We can always simplify the relation ship in the future if we really wanted to.